client/rest-high-level/src/test/java/org/opensearch/client/SearchIT.java

```diff
@@ -1827,6 +1828,40 @@ public void testCountAllIndicesMatchQuery() throws IOException {
         assertEquals(3, countResponse.getCount());
     }
 
+    public void testMatrixStatsMultiValueModeEffect() throws IOException {
```

Comment:
Considering this issue was related to cache, specifically Request Cache, I think we will also need to verify whether the first query(ie avg one) was actually cached there. And then for second query(ie min), there was a cache miss as it being treated as a different one from the first. This can be done by verifying cache stats something like [here](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/internalClusterTest/java/org/opensearch/indices/IndicesRequestCacheIT.java#L166)

Due to reasons above, I think it would be better to move this IT test inside IndicesRequestCacheIT class, until unless you think otherwise.

Reply:
Applied in 7e2e275, thanks!